<launch>
    <!-- This launch file describes the necessary params/nodes to integrate
    pose with the manipulation stack. The sole requirement that this launch file
    satisfies is that transform information must be provided between the global
    frame 'map' and the root link of the robot, 'base_footprint'.
    -->

    <arg name="live" default="true"/>
    <arg name="fake" default="false"/>
    <arg name="ns" default="false"/>

    <!-- publish fake odometry in all modes except live/real -->
    <group if="$(arg live)">
        <!--group if="$(arg fake)">
            <node name="odom_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 odom_combined base_footprint 10"/>
        </group-->
    </group>
    <group unless="$(arg live)">
        <node name="odom_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 odom_combined base_footprint 10"/>
    </group>

    <group if="$(arg live)">
        <node name="abs_nwu_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 abs_nwu map 10"/>
        <group if="$(arg fake)">
            <node name="map_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 abs_nwu base_footprint 10"/>
        </group>
    </group>
    <group unless="$(arg live)">
        <!-- to be removed: here solely to be able to use bag files with data stamped with the abs_nwu frame -->
        <node name="abs_nwu_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 abs_nwu map 10"/>

        <group if="$(arg fake)">
            <!-- fake localization -->
            <node name="map_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 abs_nwu odom_combined 10"/>
        </group>
        <group unless="$(arg fake)">
            <node name="hector_mapping" pkg="hector_mapping" type="hector_mapping">
                <rosparam command="load" file="$(find rcta)/config/localization/hector_mapping.yaml"/>
                <remap from="scan" to="base_scan"/>
            </node>
        </group>
    </group>
</launch>
